framework/src/modules/chain/logic/bft.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const assert = require('assert');
+
+const __private = {
+	items: new WeakMap(),
+};
+
+class HeadersList {
+	constructor({ size }) {
+		assert(size, 'Must provide size of the queue');
+		this.size = size;
+		__private.items.set(this, []);
+	}
+
+	get items() {
+		return __private.items.get(this);
+	}
+
+	get length() {
+		return __private.items.get(this).length;
+	}
+
+	get first() {
+		return this.items[0];
+	}
+
+	get last() {
+		return this.items[this.length - 1];
+	}
+
+	add(blockHeader) {
+		const items = this.items;
```

Comment:
these reassignment is not necessary

Reply:
Yes it is to avoid recurring calls to get items from weakmap with object reference.